### PR TITLE
Bump github.com/np-guard/models from 0.4.0 to 0.5.2

### PR DESCRIPTION
**Pull request by olasaadi99**

### CODE REVIEW

File: pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Endpoint connectivity for VPC test-vpc1-ky
 Public Internet 147.235.219.206/32 => vsi2-ky[10.240.20.4] : protocol: TCP dst-ports: 22
 db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.4] => db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5] : All Connections
-db-endpoint-gateway-ky[10.240.30.6],vsi3a-ky[10.240.30.5],vsi3b-ky[10.240.30.4] => vsi1-ky[10.240.10.4] : protocol: ICMP,UDP; protocol: TCP src-ports: 100-200 dst-ports: 10-50
```
Comment (zivnevo) [inline pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: Why this change? It makes the results far less readable

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt (hunk 1 above)
Comment (olasaadi99) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: it just replaces UDP with TCP, because of sorting

Comment (zivnevo) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: Then I would expect it to be something like:
`protocol: ICMP; protocol: TCP src-ports: 100-200 dst-ports: 10-50; protocol: UDP`

Comment (olasaadi99) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: it could be like this if we do not have the special case that Elazar added line 119 strings file

Comment (olasaadi99) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: should I remove that special case? it just calls the tcpudpstring with short version

Comment (olasaadi99) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: actually we need it for other cases like `protocol: ICMP,UDP`
without the special case it will be `protocol: ICMP; protocol: UDP` and will chane a lot of files

File: pkg/common/strings.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			return fmt.Sprintf("%s,%s", icmpString, tCPUDPString(c.TCPUDPSet(), true))
+		}
+	}
+	if tcpString != "" && icmpString != "" {
+		return fmt.Sprintf("%s%s%s", icmpString, semicolonString, tcpString)
+	}
+	return icmpString + tcpString
+}
+
+func ShortString(connections *netset.TransportSet) string {
```
Comment (zivnevo) [inline pkg/common/strings.go:130]: Unused?
Seems like it may fit in `pkg/linter/lintTCPReposnseBlocked.go:83` and in `pkg/vpcmodel/explainabilityPrint.go:638`

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/vpcmodel/detailedConn.go [hunk 3]
```diff
@@ -107,24 +108,24 @@ func (d *detailedConn) hasTCPComponent() bool {
...
 // returns the tcp responsive and non-tcp component
-func (d *detailedConn) nonTCPAndResponsiveTCPComponent() *connection.Set {
+func (d *detailedConn) nonTCPAndResponsiveTCPComponent() *netset.TransportSet {
 	return d.tcpRspEnable.Union(d.nonTCP)
 }
 
 // string adds * to non-responsive TCP components of the connection
 // for cosmetic reasons remove the protocol word from cubes prints
-func (d *detailedConn) string() string {
+func (d *detailedConn) string() string { ////////////////////olaaaaaaaaaaaaaa
```
[the remark this answers is not in the document]
Comment (olasaadi99) [reply pkg/vpcmodel/detailedConn.go:117]: removed

File: pkg/common/strings.go (hunk 2 above)
Comment (olasaadi99) [reply pkg/common/strings.go:130]: updated

File: pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt (hunk 1 above)
Comment (zivnevo) [reply pkg/ibmvpc/examples/out/analysis_out/sg_testing1_new_respond_partly_all_vpcs__with_grouping.txt:4]: Suggesting to modify the `Stringify` code as follows:
1. Identify protocols which the given `TransportSet` covers fully
2. Identify cubes which are partial to the full protocol
3. Join results from 1. and 2. together.

Something like:
```golang
func Stringify(c *netset.TransportSet, shortVersion bool) string {
	if c.IsEmpty() {
		return NoConnections
	} else if c.IsAll() {
		return AllConnections
	}

	resStrs := []string{}
	resStrs = appendIfNotEmpty(resStrs, completeProtocols(c, shortVersion))
	resStrs = appendIfNotEmpty(resStrs, iCMPString(c.ICMPSet(), shortVersion)) // partial cubes only
	resStrs = appendIfNotEmpty(resStrs, tCPUDPString(c.TCPUDPSet(), shortVersion)) // partial cubes only

	return strings.Join(resStrs, semicolonString)
}
```

Where `completeProtocols()` is something like:
```golang
func completeProtocols(c *netset.TransportSet, shortVersion bool) string {
	completeProtocols := []string{}
	if netset.AllICMPSet().IsSubset(c.ICMPSet()) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringICMP))
	}
	if netset.AllTCPTransport().IsSubset(c) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringTCP))
	}
	if netset.AllUDPTransport().IsSubset(c) {
		completeProtocols = append(completeProtocols, string(netp.ProtocolStringUDP))
	}

	res := strings.Join(completeProtocols, ",")
	if !shortVersion && res != "" {
		res = protocolString + res
	}

	return res
}
```
I believe this will not require too many changes to test results

**Review by zivnevo — APPROVED** [review]
